lib/spec/router.sw-spec.js

```diff
@@ -4,43 +4,186 @@ importScripts('/base/lib/router.js');
+    });
+
+    it('should throw Error when an extra space is provided', function () {
+      expect(function() { router.add('get ') }).to.throw(Error, 'Method "get " is not supported');
     });
   });
 
-  it('should proxy the methods to an object', function () {
-    var router = new Router();
-    var obj = {};
-    var mw = function () {};
-    sinon.spy(router, 'add');
-    router.proxyMethods(obj);
-    router.methods.forEach(function (verb) {
-      expect(obj[verb]).to.be.defined;
-    });
-    obj.get('/', mw);
-    expect(router.add.calledWith('get', '/', mw)).to.be.true;
+  describe('utility wrappers', function() {
+    it('router.methods should contain the standard HTTP verbs and "all"', function() {
+      expect(router.methods).to.contain('get');
+      expect(router.methods).to.contain('post');
+      expect(router.methods).to.contain('put');
+      expect(router.methods).to.contain('delete');
+      expect(router.methods).to.contain('head');
+      expect(router.methods).to.contain('all');
+      expect(router.methods.length).to.be.equal(6);
+    });
+
+    it('should have defined methods for available verbs', function() {
+      router.methods.forEach(function (verb) {
+        expect(router[verb]).to.be.defined;
+      });
+    });
+
+    it('should wrap add()', function() {
+      var handler = function() {};
+      sinon.spy(router, 'add');
+      router.get('/', handler);
+      expect(router.add.calledWith('get', '/', handler)).to.be.true;
+    });
   });
 
-  it('should recover all the middlewares attached to a route and a method',
-  function () {
-    var router = new Router();
-    var mw1 = function () {};
-    var mw2 = function () {};
-    var mw3 = function () {};
-    router.get('/a', mw1);
-    router.get('/a', mw2);
-    router.post('/a', mw3);
-    router.get('/b', mw3);
-
-    expect(router.match('get', '/a')).to.deep.equal([mw1, mw2]);
-    expect(router.match('post', '/a')).to.deep.equal([mw3]);
-    expect(router.match('get', '/b')).to.deep.equal([mw3]);
+  describe('proxyMethods()', function() {
+    var object;
+    var handler = function() {};
+
+    beforeEach(function() {
+      object = {};
+    });
+
+    it('should proxy the methods to an object', function () {
+      router.proxyMethods(object);
+      router.methods.forEach(function (verb) {
+        expect(object[verb]).to.be.defined;
+      });
+    });
+
+    it('should wrap add()', function() {
+      sinon.spy(router, 'add');
+      router.proxyMethods(object);
+      object.get('/', handler);
+      expect(router.add.calledWith('get', '/', handler)).to.be.true;
+    });
+
+    // Currently failing, due to a testability issue: the handler gets binded,
+    // which prevents to test the equality of the handlers.
+    // If you guys have any idea on how to improve that.
+    it('should support an object as a middleware', function() {
```

Comment:
I broke down this test into 3:
* should wrap add()
* should call onFetch()
* onFetch() should bind "this" to the handler